From e9d22a6f30941cad86b22258fd0a467a7b69ede4 Mon Sep 17 00:00:00 2001
From: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Date: Mon, 31 Jul 2017 19:26:05 +0300
Subject: [PATCH 026/122] media: soc_camera: Fix VIDIOC_S_SELECTION ioctl
 miscalculation

This patch corrects the miscalculation of the capture buffer
size and clipping data update in VIDIOC_S_SELECTION sequence.

Patch isbased on work by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>

Signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
---
 drivers/media/platform/soc_camera/soc_scale_crop.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/media/platform/soc_camera/soc_scale_crop.c b/drivers/media/platform/soc_camera/soc_scale_crop.c
index 092c73f..6050f52 100644
--- a/drivers/media/platform/soc_camera/soc_scale_crop.c
+++ b/drivers/media/platform/soc_camera/soc_scale_crop.c
@@ -126,6 +126,7 @@ int soc_camera_client_s_selection(struct v4l2_subdev *sd,
 		dev_dbg(dev, "Camera S_SELECTION successful for %dx%d@%d:%d\n",
 			rect->width, rect->height, rect->left, rect->top);
 		*target_rect = *cam_rect;
+		*subrect = *rect;
 		return 0;
 	}
 
@@ -217,6 +218,7 @@ int soc_camera_client_s_selection(struct v4l2_subdev *sd,
 
 	if (!ret) {
 		*target_rect = *cam_rect;
+		*subrect = *rect;
 		move_and_crop_subrect(target_rect, subrect);
 	}
 
@@ -297,9 +299,7 @@ static int client_set_fmt(struct soc_camera_device *icd,
 	if (ret < 0)
 		return ret;
 
-	if (host_1to1)
-		*subrect = *rect;
-	else
+	if (!host_1to1)
 		move_and_crop_subrect(rect, subrect);
 
 	return 0;
-- 
2.7.4

